common/resource/fx.go

```diff
@@ -195,6 +198,16 @@ func NamespaceRegistryProvider(
 	)
 }
 
+func ConfigProvider(
```

Comment:
If this is here (and it shouldn't be), it needs a new name, "ConfigProvider" for something that provides _history_ configs is too generic outside of the context of the history service.